src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  };
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && prey.hidden === false) {
+      prey.health -= 50;
+      prey.health === 0 && Animal.alive.splice(Animal.alive.indexOf(prey), 1);
```

Comment:
That's not readable, you would suggest the following changes ⬇️ 
```suggestion
            if (!prey.health) {
        const preyIndex= Animal.alive.indexOf(prey);

        Animal.alive.splice(preyIndex, 1);
```